flink-runtime/src/main/java/org/apache/flink/runtime/leaderelection/DefaultLeaderElection.java

```diff
@@ -56,4 +56,40 @@ public boolean hasLeadership(UUID leaderSessionId) {
     public void close() throws Exception {
         parentService.remove(contenderID);
     }
+
+    /**
+     * {@link ParentService} is introduced to define the protocol between {@link
+     * LeaderElectionService} and {@link LeaderElection}.
+     */
+    abstract static class ParentService {
+
+        /**
+         * Registers the {@link LeaderContender} under the {@code contenderID} with the underlying
+         * {@code LeaderElectionService}. Leadership changes are starting to be reported to the
```

Comment:
```suggestion
         * {@code ParentService}. Leadership changes are starting to be reported to the
```
I guess we have to rename `LeaderElectionService` into `ParentService` in the JavaDoc as well. ...since we're not implementing the interface anymore.

This applies to other JavaDoc comments in this class as well.

Reply:
Done.